roles/nginx/tasks/main.yml

```diff
@@ -16,9 +16,9 @@
     src: "{{ item.src}}"
     dest: "{{ item.dest }}"
   with_items:
-    - { src: 'server.conf', dest: '/etc/nginx/' }
-    - { src: 'nginx.conf', dest: '/etc/nginx/' }
-    - { src: 'ports.conf', dest: "/etc/{{ apache_service }}/" }
+    - { src: 'server.conf.j2', dest: '/etc/nginx/' }
+    - { src: 'nginx.conf.j2', dest: '/etc/nginx/' }
+    - { src: 'ports.conf.j2', dest: "/etc/{{ apache_service }}/" }
```

Comment:
need to use the file name when writing the file name.
dest ... /etc/nginx/server.conf
dest ... /etc/nginx/nginx.conf
dest ... /etc/{{ apache_service }}/ports.conf